src/stirling/source_connectors/socket_tracer/socket_trace_connector.cc

```diff
@@ -176,7 +176,7 @@ DEFINE_bool(
     stirling_debug_tls_sources, gflags::BoolFromEnv("PX_DEBUG_TLS_SOURCES", false),
     "If true, stirling will add additional prometheus metrics regarding the traced tls sources");
 
-DEFINE_uint32(stirling_bpf_loop_limit, 42,
+DEFINE_uint32(stirling_bpf_loop_limit, 41,
```

Comment:
The iovec syscall return probes that rely on process_implicit_conn (recvmsg, recvmmsg, sendmsg, sendmmsg) had their instruction count increase beyond the 4.14 limit. This was my solution to getting the BPF instruction count on 4.14 kernels to work. The rationale is that cases that are right on the edge of this limit are likely to have problems already.

The other things I considered were the following:
- Remove match_trace_tgid logic on 4.14 kernels -- did not lower the instruction count enough
- Make the iovec ret syscalls provide a NULL sock for its `submit_new_conn` call (e.g. `submit_new_conn(ctx, tgid, fd, addr, /*sock*/ NULL, role, source_fn)`) -- not viable since it partially handles mid stream connections

I'm open to other suggestions if you have any, but this was the set of things I came up with.